geode-cq/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/DurableClientSimpleDUnitTest.java

```diff
@@ -390,81 +327,33 @@ public void testDurableClientMultipleServersOneLive() {
-      @Override
-      public void run2() throws CacheException {
-        CacheServerTestUtil.getClientCache().readyForEvents();
-      }
-    });
+    sendClientReady(this.durableClientVM);
 
-    // Have the durable client register interest in all keys
-    this.durableClientVM.invoke(new CacheSerializableRunnable("Register interest") {
-      @Override
-      public void run2() throws CacheException {
-        // Get the region
-        Region region = CacheServerTestUtil.getCache().getRegion(regionName);
-        assertNotNull(region);
-
-        // Register interest in all keys
-        region.registerInterestRegex(".*", InterestResultPolicy.NONE, true);
-      }
-    });
+    registerInterest(durableClientVM, regionName, true, NONE);
 
     // Verify durable client on server1
-    this.server1VM.invoke(new CacheSerializableRunnable("Verify durable client") {
-      @Override
-      public void run2() throws CacheException {
-        // Find the proxy
-        checkNumberOfClientProxies(1);
-        CacheClientProxy proxy = getClientProxy();
-        assertNotNull(proxy);
-
-        // Verify that it is durable and its properties are correct
-        assertTrue(proxy.isDurable());
-        assertEquals(durableClientId, proxy.getDurableId());
-        assertEquals(durableClientTimeout, proxy.getDurableTimeout());
-      }
-    });
-
+    verifyDurableClientPresent(durableClientTimeout, durableClientId, server1VM);
     // Start normal publisher client
-    this.publisherClientVM.invoke(() -> CacheServerTestUtil
-        .createCacheClient(getClientPool(getServerHostName(),
-            server1Port, server2Port, false), regionName));
+    this.publisherClientVM.invoke(() -> createCacheClient(getClientPool(getServerHostName(),
+        server1Port, server2Port, false), regionName));
 
     // Publish some entries
     final int numberOfEntries = 10;
-    publishEntries(numberOfEntries);
+    publishEntries(0, numberOfEntries);
 
     // Verify the durable client received the updates
-    this.durableClientVM.invoke(new CacheSerializableRunnable("Verify updates") {
-      @Override
-      public void run2() throws CacheException {
-        // Get the region
-        Region region = CacheServerTestUtil.getCache().getRegion(regionName);
-        assertNotNull(region);
+    checkListenerEvents(numberOfEntries, 1, -1, this.durableClientVM);
 
-        // Get the listener and wait for the appropriate number of events
-        CacheServerTestUtil.ControlListener listener =
-            (CacheServerTestUtil.ControlListener) region.getAttributes().getCacheListeners()[0];
-        listener.waitWhileNotEnoughEvents(30000, numberOfEntries);
-        assertEquals("Events were" + listener.events, numberOfEntries, listener.events.size());
-      }
-    });
-
-    try {
-      Thread.sleep(10000);
-    } catch (InterruptedException ex) {
-      fail("interrupted", ex);
-    }
+    sleep(10000);
```

Comment:
This is a magic timeout. Remove it and things fail. Add it in and things are happy. I have no idea what this is really affecting, but it is necessary to work properly.

Reply:
This test appears to be ignored. Also it fails on my machine (when run with IntelliJ). Maybe a good future task.